src/sqlfluff/core/parser/segments/base.py

```diff
@@ -804,6 +812,13 @@ def to_tuple(self, code_only=False, show_raw=False, include_meta=False):
             )
         return result
 
+    def copy(self):
```

Comment:
Could we have used `copy.deepcopy()` here?

Reply:
Tried that - unfortunately `copy.copy()` doesn't copy enough and `copy.deepcopy()` copies too much.